src/main/java/hudson/plugins/promoted_builds/Promotion.java

```diff
@@ -87,6 +90,13 @@ public Status getStatus() {
         return getTarget().getAction(PromotedBuildAction.class).getPromotion(getParent().getName());
     }
 
+	public String getIcon() {
+		return icon;
+	}
+
+	public void setIcon(String icon) {
```

Comment:
Yes, firstly it makes sense to save the promotion after such change. Another concern is the setter method at all. IMO it would be preferable to replace the entire <code>Promotion</code> by a new instance like it is being done in configuration pages.